### DISCUSSION TITLE
Mysql version 8.4 (LTS), 9.0 and 9.1 (Innovation) for Laragon 6

### DISCUSSION THREAD

**Discussion by borhanuddin** [leokhoa/laragon#932] [Laragon's Articles & How-tos]
This guide is to help those who struggled to use MySQL version 8.4 and above.
This simplified guide has been created based on reading the below articles (credits to the owner):

- https://github.com/leokhoa/laragon/discussions/822
- https://github.com/leokhoa/laragon/issues/772
- https://gist.github.com/meorajrul/b57803bf1b4ddfd2f93e6ad37c3ac5f2

**WARNING! Please back up/dump all your databases. I'm not responsible for any data loss by following this guide.**

_In this example, I'm using MySQL version 9.1 as a reference._

1. Download and extract the MySQL zip file to the `C:\laragon\bin\mysql` folder.
2. Right-click on Laragon Tray and switch to MySQL 9.1 version.
3. Start MySQL (this will initialise MySQL data and produce an error message; ignore the error and click OK). We do this just to create a `my.ini` file, which can be located in `C:\laragon\bin\mysql\mysql-9.1.0-winx64`.
4. Open `my.ini` file.
5. **REMOVE** the line `default-authentication-plugin=mysql_native_password` from the file, save it, and close the file.
6. Set the file `my.ini` as **READ-ONLY** (to avoid Laragon overriding settings using the old template).
7. Navigate to `C:\laragon\data` and **DELETE** folder `mysql-9` (this folder is being created when we first start the MySQL in step no. 3).
8. Start MySQL again.

MySQL will start successfully.

If you want to use HeidiSQL as your database client software, make sure you check the "Prompt for credentials" option under Hostname/IP. A credential pop-up box will appear. Just click the LOGIN button without any password.

I hope that people who needed this guide found it useful.

**Comment by FlavioSuarez** [comment]
Warning!!
On MySQL 9.0, the mysql_native_password plugin was removed completely.
Before upgrading MySQL, change the authentication plugin of the MySQL users (this includes the root user) to caching_sha2_password.
More info at
https://php.watch/articles/fix-php-mysql-84-mysql_native_password-not-loaded

**Comment by Ringoinmel** [comment]
It doesn't work and the screen shows initializing for 10 minutes and then I need to stop it

**Reply by AWqxKAWERbXo** [reply]
Did you try initialize by console first, e.g. via PowerShell?

```
mysqld --initialize --datadir="C:\laragon-6.0.0\data\mysql-9" --basedir="C:\laragon\bin\mysql\mysql-9.0.0-winx64" --console
```

**Comment by borhanuddin** [comment]
Just want to update the above guide: when I upgrade to MySQL 9.3.0 from 9.1.0, just follow steps 1-6.
